Additions to author and reviewer guidance, and language code review checklists (#582)
microsoft/code-with-engineering-playbook: code-reviews/process-guidance/reviewer-guidance.md

- Be positive – encouraging, appreciation for good practices.
- Avoid language that points fingers like “you” but rather use “we” or “this line” -- code reviews are not personal and language matters.
- Prefer asking questions above making statements. There might be a good reason for the author to do something.
- If you make a direct comment, explain why the code needs to be changed, preferably with an example.
- If a few back-and-forth comments don't resolve a disagreement, have a quick talk with eachother (in-person or call). Don't forget to update the PR with what you agreed on and why.

### Make comments clear

- Explain why the code needs to change.
- Prefix a “point of polish” with “Nit:” - or use the Azure DevOps `suggestion` button and propose the correction directly.

> - If one or two comments don’t resolve a disagreement, talk in person or on the phone.

Reviewer: Github also has this option  - may be it could be rephrased more broadly?

Reviewer: actually i did as we recommend here:
I did 2 things: 1. committed suggestion directly to your branch and 2. created PR to your branch.  
I think 2(PR way) is preferable. 
I however do not see "accept changes" directly in PR that i have seen before.

Author: Good point, thank you! I think that your changes are now reflected in this PR, so all good :).

Reviewer 2: I missed this conversation - @zoekdestep please ignore my overlapping/conflicting suggestion

- Be positive – encouraging, appreciation for good practices.
- Avoid language that points fingers like “you” but rather use “we” or “this line” -- code reviews are not personal and language matters.
- Prefer asking questions above making statements. There might be a good reason for the author to do something.
- If you make a direct comment, explain why the code needs to be changed, preferably with an example.
- If a few back-and-forth comments don't resolve a disagreement, have a quick talk with eachother (in-person or call). Don't forget to update the PR with what you agreed on and why.

### Make comments clear

- Explain why the code needs to change.

> - Prefix a “point of polish” with “Nit:” 

Reviewer 2: ```suggestion
- Prefix a “point of polish” with “Nit:”.
```

> - Propose suggestion by creating PR to the author branch if feasible.

Reviewer 2: ```suggestion
- Suggest changes to a PR by using the suggestion feature (available in [GitHub](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request) and Azure DevOps) or by creating a PR to the author branch.
```

> - If one or two comments don’t resolve a disagreement, talk in person or create group discussion

Reviewer 2: ```suggestion
- If one or two comments don’t resolve a disagreement, talk in person or create group discussion.
```
